crates/sui-core/src/authority/authority_store.rs

```diff
@@ -807,30 +745,11 @@ impl AuthorityStore {
             .cloned()
             .collect();
 
-        // Index the certificate by the objects mutated
-        write_batch = write_batch.insert_batch(
-            &self.perpetual_tables.parent_sync,
-            written
-                .iter()
-                .map(|(_, (object_ref, _object, _kind))| (object_ref, transaction_digest)),
-        )?;
-
-        // Index the certificate by the objects deleted
         write_batch = write_batch.insert_batch(
-            &self.perpetual_tables.parent_sync,
-            deleted.iter().map(|(object_id, (version, kind))| {
-                (
-                    (
-                        *object_id,
-                        *version,
-                        if kind == &DeleteKind::Wrap {
-                            ObjectDigest::OBJECT_DIGEST_WRAPPED
-                        } else {
-                            ObjectDigest::OBJECT_DIGEST_DELETED
-                        },
-                    ),
-                    transaction_digest,
-                )
+            &self.perpetual_tables.objects,
+            deleted.iter().map(|(object_id, (version, _))| {
+                let tombstone: StoreObjectWrapper = StoreObject::Deleted.into();
+                (ObjectKey(*object_id, *version), tombstone)
```

Comment:
i think we will need to be able to distinguish between wraps and deletes. this code is intended to detect deletes of wrapped objects: https://github.com/MystenLabs/sui/blob/main/crates/sui-adapter/src/programmable_transactions/context.rs#L611-L623

it will currently work because it doesn't care about the tombstone type. but, i think we should be able to prune deleted tombstones, and preserve only wrapped tombstones.

Reply:
I don't think we need to store them differently. At least I don't see a use case that requires that so far. 
Pruner also works based on checkpoint effects and receives DeletionKind

Reply:
added Wrapped variant to the enum